lib/org/chromium/webidl/URLExtractor.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'URLExtractor',
+
+  requires: ['org.chromium.webidl.IDLFileContents'],
+
+  properties: [
+    {
+      name: 'excludeRegexp',
+      // documentation: 'Excludes all URLs matching this expression.',
+    },
+    {
+      name: 'includeRegexp',
+      // documentation: 'Only include URLs matching this expression.',
+    }
+  ],
+
+  constants: {
+    URL_REGEX: /https?:\/\/(www\.)?[-_a-zA-Z0-9.]{2,256}\.[a-z]{2,6}(\/[^?#, \n]*)?(\?[^#, \n]*)?/g,
```

Comment:
The original pattern doesn't know how to deal with a domain-only URL. This probably works though:
```
https?:\/\/[^/?#, \n]+(\/[^?#, \n]*)?(\?[^#, \n]*)?
```
Though now that I've seen it, I think I like this even better:
```
https?:\/\/[^#, \n]
```
because delegating the decision of what is and isn't a valid URL to some other code feels like the right thing to do.